controllers/flowcollector_controller.go

```diff
@@ -352,15 +450,17 @@ func (r *FlowCollectorReconciler) newClientHelper(desired *flowsv1alpha1.FlowCol
 func (r *FlowCollectorReconciler) failure(ctx context.Context, errcond *conditions.ErrorCondition, fc *flowsv1alpha1.FlowCollector) error {
 	log := log.FromContext(ctx)
 	log.Error(errcond.Error, errcond.Message)
-	meta.SetStatusCondition(&fc.Status.Conditions, errcond.Condition)
-	if errUpdate := r.Status().Update(ctx, fc); errUpdate != nil {
-		log.Error(errUpdate, "Set conditions failed")
-	}
+	r.updateCondition(ctx, &errcond.Condition, fc)
 	return errcond.Error
 }
 
 func (r *FlowCollectorReconciler) updateCondition(ctx context.Context, cond *metav1.Condition, fc *flowsv1alpha1.FlowCollector) {
-	meta.SetStatusCondition(&fc.Status.Conditions, *cond)
+	r.updateConditions(ctx, &[]metav1.Condition{*cond}, fc)
+}
+
+func (r *FlowCollectorReconciler) updateConditions(ctx context.Context, conds *[]metav1.Condition, fc *flowsv1alpha1.FlowCollector) {
+	conditions.SetNewConditions(&fc.Status.Conditions, conds)
+
 	if err := r.Status().Update(ctx, fc); err != nil {
 		log.FromContext(ctx).Error(err, "Set conditions failed")
 		// Do not propagate this update failure: if update failed it's generally because it was modified concurrently:
```

Comment:
@jotak have you tried to use `patch` here instead ? 

Since I have implemented the watchers, I get more concurrent modifications than before and it miss some conditions updates.
The worst situation would be missing `namespace` update here since it's stored in the `Status` and we update everything once.

Reply:
no I haven't tried patch, I guess it's worth investigating